File: solutions/go/resistor-color/1/resistor_color.go
package resistorcolor

var resistor = map[string]int{
	"black":  0,
	"brown":  1,
	"red":    2,
	"orange": 3,
	"yellow": 4,
	"green":  5,
	"blue":   6,
	"violet": 7,
	"grey":   8,
	"white":  9,
}

// Colors should return the list of all colors.
func Colors() []string {
	result := []string{}
	for c := range resistor {
		result = append(result, c)
	}
	return result
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	return resistor[color]
}

File: solutions/cpp/resistor-color/1/resistor_color.cpp
#include "resistor_color.h"
#include <algorithm>
#include <iterator>

namespace resistor_color {
int color_code(std::string const &in1) {
  return std::distance(cc.begin(), std::find(cc.begin(), cc.end(), in1));
}

std::vector<std::string> colors() {
  return cc;
}
} // namespace resistor_color
